spx-gui/src/class/asset-list.ts

```diff
@@ -1,27 +1,49 @@
-import { AssetBase } from "./asset-base";
-import { Sound } from "./sound";
-import { Sprite } from "./sprite";
+import { AssetBase } from './asset-base'
+import { Project } from './project'
+import { Sound } from './sound'
+import { Sprite } from './sprite'
 
 export abstract class AssetList<T extends AssetBase> {
-    public list: T[] = [];
+  public list: T[] = []
 
-    constructor(list: T[] = []) {
-        this.list = list;
-    }
+  constructor(list: T[] = []) {
+    this.list = list
+  }
 
-    add(...assets: T[]) {
-        this.list.push(...assets);
-    }
+  add(...assets: T[]) {
+    this.list.push(...assets)
+  }
 
-    remove(name: string): T | null;
-    remove(item: T): T | null;
+  remove(name: string): T | null
+  remove(item: T): T | null
 
-    remove(item: string | T): T | null {
-        const index = this.list.findIndex(i => typeof item === 'string' ? i.name === item : i === item);
-        return index > -1 ? this.list.splice(index, 1)[0] : null;
-    }
+  remove(item: string | T): T | null {
+    const index = this.list.findIndex((i) =>
+      typeof item === 'string' ? i.name === item : i === item
+    )
+    return index > -1 ? this.list.splice(index, 1)[0] : null
+  }
 }
 
-export class SpriteList extends AssetList<Sprite> { }
+export class SpriteList extends AssetList<Sprite> {
+  project: Project
+  constructor(project: Project) {
+    super()
+    this.project = project
+  }
+  addSpriteToProject(sprite: Sprite): void {
+    if (sprite) {
+      this.add(sprite)
+      this.project.backdrop.config.zorder.push(sprite.name)
+    }
+  }
+  removeSpriteFromProject(sprite: Sprite): void {
+    const index = this.project.backdrop.config.zorder.indexOf(sprite.name)
+    if (index !== -1) {
+      this.remove(sprite)
+      this.project.backdrop.config.zorder.splice(index, 1)
+    }
+  }
+}
```

Comment:
Since the operation of adding sprites and updating zorder is required not only in spriteStore, but also in adding default sprites, the operation of adding sprites and updating zorder is placed in SpriteList